Fix parsing of multivalued list type in client mappers

Wareload opened the pull request:
Enable stringify by default, backend expects only a string and currently errors on parsing

Closes #26794

a list was sent instead of a string:
![image](https://github.com/keycloak/keycloak/assets/74083411/7269d2fd-f192-4261-806b-e6377f83723a)
![image](https://github.com/keycloak/keycloak/assets/74083411/4f801ca6-000a-4a83-95be-8b1ba93d6d9a)


<!---
Please read https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md and follow these guidelines when contributing to Keycloak
-->

jonkoops reviewed (changes requested):
This is very likely to cause issues where now things will break where `stringify` is not expected to be the default.

Wareload committed:
Fix parsing of multivalued list type

Enable stringify by default, backend expects only a string and currently errors on parsing

Closes #26794

Signed-off-by: Daniel Haas <daniel.haas@wareload.net>

jonkoops commented on js/libs/keycloak-admin-client/src/defs/configPropertyRepresentation.ts:
```diff
@@ -10,4 +10,5 @@ export interface ConfigPropertyRepresentation {
   options?: string[];
   secret?: boolean;
   required?: boolean;
+  stringify?: boolean;
```
Properties in representations of the admin client should only be added if the fields actually exist on the representation on the Java-side, which this does not.

Wareload replied:
The core issue here is, that all ConfigProperties of ClientMappers must be strings and therefore should be stringified. I have multiple solutions in mind, what is you opinion on that?

- Hardcode the stringify attribute in the frontend, even though it does not exist in the interface in the backend. Because some elements use it, it should be part of the configPropertyRepresentation. 
- Extend the representation in the frontend and backend with the stringify attribute, to support setting it manuelly in the mappers
- Change the expected request object on the backend side from Map<String,String> to Map<String,Object>

From my current understanding, solution two would be the appropriate solution, right?

jonkoops replied:
Yeah this is a hard one, I think ideally the backend determines which fields should be stringified, but I don't believe this is accurately done or consistently for that manner.

Wareload replied:
I added it to the backend as well, so there is at least the possibilty to define and use it

Wareload committed:
Fix parsing of multivalued list type

Closes #26794

Signed-off-by: Daniel Haas <daniel.haas@wareload.net>

Wareload commented:
I added the stringify attribute and the corresponding logic to the backend

hmlnarik reviewed (changes requested):
Thank you for the PR. I am not clear on how the `stringify` property in the server is supposed to work though. It only seems to have meaning if the property type is `MULTIVALUED_LIST_TYPE` and no meaning for another property types.

Can you clarify the use with some examples? Is there a case when one would _not_ stringify multivalued list?

For having this PR finalized, a test validating the new functionality would be needed as well.

Wareload commented:
My use case would be a custom oidc mapper, which uses the MULTIVALUED_LIST_TYPE(with explizit setting the stringify value to true). Therefore it must be always stringified because the backend is expecting a string. For compatibility reasons I have not set it generally to true, because I am not aware of side effects by changing the default behaviour and other use cases (on maybe other components) and this fix allows the MULTIVALUED_LIST_TYPE to be used again if wanted.

edewit commented:
closed in favour of #32224
